lib/fuser_mqubit.h

```diff
@@ -955,7 +955,8 @@ class MultiQubitGateFuser final : public Fuser<IO, Gate> {
                   if (ln != nullptr && rn != nullptr) {
                     return R()(ln->val->parent->time, rn->val->parent->time);
                   } else {
-                    return ln != nullptr || rn == nullptr;
+                    // Comparator needs to be irreflexive to get a strict weak ordering.
+                    return ln != nullptr && rn == nullptr;
```

Comment:
I think `&& rn == nullptr` is redundant here because the case `ln != nullptr && rn != nullptr` is checked in line 995.

Reply:
Oooooh, finally got it, thanks, Sergei!